mne/source_estimate.py

```diff
@@ -2860,6 +2860,16 @@ def _temporary_vertices(src, vertices):
             s['vertno'] = v
 
 
+def _check_stc_src(stc, src):
+    if stc is not None and src is not None:
+        for s, v, hemi in zip(src, stc.vertices, ('left', 'right')):
+            n_missing = (~np.in1d(v, s['vertno'])).sum()
+            if n_missing:
+                raise ValueError('%d/%d %s hemisphere stc vertices '
+                                 'missing from the source space, likely '
+                                 'mismatch' % (n_missing, len(v), hemi))
```

Comment:
what you should check is that src contains a superset of the vertices in stc as it's all you need. You need in the src the vertices from stc but if you have more it's ok.

Reply:
I think that's what's already done here already with the (effectively) `(~np.in1d(stc.vertices[ii], src[ii]['vertno'])).sum()`, i.e., "how many of the stc vertices are not in the source space". So nothing to change here I think?

... except that it would be good to check not just the surface STC + surface source but also volume STC + volume source. The same check applies there. Personally I would just check that `stc.vertices` and `src` are the same length, then iterate directly over the `stc.vertices` and `src` and just say that `stc.vertices[{ii}]` has a problem. But if it's already split into surfaces and volume somewhere then we could allow it to be more specific.